### ISSUE TITLE
Provide a iterative topological sort algorithm?

### ISSUE THREAD

**Issue by harlowja** [networkx/networkx#1751]
To match most of the other iterative algorithms it would be great if there could also be a iterative topological sort algorithm as well that would iteratively yield nodes back.

**Comment by harlowja** [comment]
An example where this would be actually useful:

https://github.com/openstack/taskflow/blob/master/taskflow/patterns/graph_flow.py#L268

In that function, since its an iterative version, having a matching iterative topological sort (if possible) would make sense IMHO

**Comment by MisterSheik** [comment]
I've already written this, but I'm too busy with work right now to push it.  https://github.com/networkx/networkx/pull/1662

I'll try tomorrow if you can help me with the rebase.

**Comment by harlowja** [comment]
Sure, will see if I can rebase that for u.

**Comment by MisterSheik** [comment]
oh, you don't have to do it for me.  I just want to know how to do it so that I know for next time :)

**Comment by harlowja** [comment]
Oh, ha.

Well this is what I usually do, I pretty sure there are about 500 other ways of doing the same thing.

```
# perform this on git cloned fork
$ git remote add upstream git@github.com:networkx/networkx.git
# ensure remote is listed
$ git remote
origin
upstream
$ git fetch upstream
$ git checkout <your-special-branch>
$ git rebase upstream/master
# resolve conflicts, and then when done push up <your-special-branch>; it will now be synced with upstream/master
```

That should do it.

**Comment by MisterSheik** [comment]
Ok, I'll try that now.  If you don't mind, I'll post any problems here.

**Comment by MisterSheik** [comment]
(and thanks)

**Comment by harlowja** [comment]
Sure :)

**Comment by harlowja** [comment]
If u need help, I just made ``#networkx`` channel on freenode, so I'll be there :-P

**Comment by harlowja** [comment]
(don't think there is any other network IRC channel that I know about)

**Comment by MridulS** [comment]
we did have a GSoC 2015 specific gitter channel (kind of online irc)

**Comment by MridulS** [comment]
https://gitter.im/networkx/GSoC_2015

**Comment by harlowja** [comment]
Woah, I'm old-school I guess, ha. All the other [openstack](openstack.org) stuff I'm involved with uses freenode, so that's where I hang ;)

**Comment by harlowja** [comment]
irc://chat.freenode.net/networkx ftw :-P

**Comment by harlowja** [comment]
(it is sorta lonely there though, ha)

**Comment by MisterSheik** [comment]
okay, I need to find an irc client…

**Comment by MridulS** [comment]
https://webchat.freenode.net

**Comment by harlowja** [comment]
A few hundred @ https://en.wikipedia.org/wiki/Comparison_of_Internet_Relay_Chat_clients ;)

IMHO http://hexchat.github.io/ is pretty nice, and cross-platform.

**Comment by dschult** [comment]
I believe this issue is now fixed with #1756 merged.  
Anything else here?

**Comment by MisterSheik** [comment]
@dschult  I think this is safe to close.
